core/templates/dev/head/pages/exploration_editor/simple_editor_tab/SimpleEditorSidebarDirective.js

```diff
@@ -21,12 +21,37 @@ oppia.directive('simpleEditorSidebar', [function() {
     restrict: 'E',
     templateUrl: 'simpleEditor/sidebar',
     controller: [
-        '$scope', 'EditorModeService', 'SimpleEditorManagerService',
-        function($scope, EditorModeService, SimpleEditorManagerService) {
+       '$scope', 'EditorModeService', 'SimpleEditorManagerService',
+        'ScrollSyncService', 'QuestionHashService',
+        function($scope, EditorModeService, SimpleEditorManagerService,
+                 ScrollSyncService, QuestionHashService) {
       $scope.SUBFIELD_LABELS = [
         'Multiple choice', 'Correct answer', 'Hints', 'Bridge text'];
       $scope.setEditorModeToFull = EditorModeService.setModeToFull;
       $scope.questionList = SimpleEditorManagerService.getQuestionList();
+      $scope.getSidebarItemHash = function(question, subfieldLabel) {
+        return QuestionHashService.getSidebarItemHash(
+          question.getStateName(), subfieldLabel
+        );
+      };
+      $scope.scrollToField = function(question, subfieldLabel) {
+        ScrollSyncService.scrollTo(
+          QuestionHashService.getSubfieldHash(
+            question.getStateName(), subfieldLabel
+          )
+        );
+      };
+      $scope.scrollToHeader = ScrollSyncService.scrollTo;
+      $scope.toggleCollapse = function(question) {
+        question.collapsed = !question.collapsed;
+      };
+      $scope.$on('SimpleEditorSidebarToggleCollapse', function(evt, question) {
+        $scope.toggleCollapse(question);
```

Comment:
Hmm. If this is all this does, why not just call question.toggleCollapse() instead of doing the $broadcast, and thus remove the need for the event?